src/formatter/terminal_themed.rs

```diff
@@ -0,0 +1,76 @@
+use std::cell::RefCell;
+use std::io::Write;
+use std::sync::Arc;
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+use super::{Formatter, HighlightEvent, Theme};
+use crate::constants::HIGHLIGHT_NAMES;
+use crate::Result;
+
+/// A formatter for highlighting into terminal output with ANSI color codes.
+///
+/// - The primary color of each style is used for the text color.
+/// - The secondary color is not used in this formatter.
+///
+/// This formatter holds theme data in an [`Arc<Theme>`], making it cheap to clone and usable
+/// from multiple threads.
+#[derive(Debug)]
+pub struct TerminalFormatter {
+    theme: Arc<Theme>,
+    stdout: RefCell<StandardStream>,
+}
+
+impl TerminalFormatter {
+    /// Create a new [`TerminalFormatter`] with the given theme.
+    pub fn new(theme: Theme) -> Self {
+        Self {
+            theme: Arc::new(theme),
+            stdout: RefCell::new(StandardStream::stdout(ColorChoice::Always)),
```

Comment:
I wouldn't assume that users want standard output - consider something like printing code snippets in a backtrace to standard error. Maybe take a `StandardStream` as one of the parameters.